auth/src/main/java/com/firebase/ui/auth/ui/idp/AuthMethodPickerActivity.java

```diff
@@ -229,14 +228,32 @@ private void populateIdpListCustomLayout(List<IdpConfig> providerConfigs) {
         for (IdpConfig idpConfig : providerConfigs) {
             final String providerId = providerOrEmailLinkProvider(idpConfig.getProviderId());
 
-            if (!providerButtonIds.containsKey(providerId)) {
+            Integer buttonResId = providerButtonIds.get(providerId);
+            if (buttonResId == null) {
                 throw new IllegalStateException("No button found for auth provider: " + idpConfig.getProviderId());
             }
 
-            @IdRes int buttonId = providerButtonIds.get(providerId);
+            @IdRes int buttonId = buttonResId;
             View loginButton = findViewById(buttonId);
             handleSignInOperation(idpConfig, loginButton);
         }
+        //hide custom layout buttons that don't have their identity provider set
+        for (String providerBtnId : providerButtonIds.keySet()) {
+            if (providerBtnId == null) continue;
```

Comment:
nit: we try not to use single-line if statements
```suggestion
            if (providerBtnId == null) {
                continue;
            }
```

Reply:
Done.